src/main/java/io/cdap/plugin/gcp/gcs/actions/GCSBucketCreate.java

```diff
@@ -127,17 +131,22 @@ public void run(ActionContext context) throws Exception {
           bucket = storage.get(gcsPath.getBucket());
         } catch (StorageException e) {
           // Add more descriptive error message
-          throw new RuntimeException(
-            String.format("Unable to access or create bucket %s. ", gcsPath.getBucket())
-              + "Ensure you entered the correct bucket path and have permissions for it.", e);
+          String errorReason = String.format("Unable to access or create bucket %s. ", gcsPath.getBucket()) +
+            "Ensure you entered the correct bucket path and have permissions for it.";
+          String errorMessage = String.format("%s %s", errorReason, e.getMessage());
+          throw ErrorUtils.getProgramFailureException(new ErrorCategory(ErrorCategory.ErrorCategoryEnum.PLUGIN),
+            errorReason, errorMessage, ErrorType.USER, true, e);
         }
         if (bucket == null) {
           GCPUtils.createBucket(storage, gcsPath.getBucket(), config.location, cmekKeyName);
           undoBucket.add(bucketPath);
         } else if (gcsPath.equals(bucketPath) && config.failIfExists()) {
           // if the gcs path is just a bucket, and it exists, fail the pipeline
           rollback = true;
-          throw new Exception(String.format("Path %s already exists", gcsPath));
+          String errorReason = String.format("Path %s already exists", gcsPath);
+          throw ErrorUtils.getProgramFailureException(new ErrorCategory(ErrorCategory.ErrorCategoryEnum.PLUGIN),
+            errorReason, errorReason, ErrorType.SYSTEM, true, null);
```

Comment:
this is not a `system` error, it is a `user` error as bucket is provided by user.